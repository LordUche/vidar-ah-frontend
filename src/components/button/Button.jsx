import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

const Button = ({ text }) => (
  <button className="yellow-button">{text}</button>
);

Button.propTypes = {
  text: PropTypes.string.isRequired
};

export default Button;
