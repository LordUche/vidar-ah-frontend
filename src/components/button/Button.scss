.yellow-button {
  background: #e2d754;
  border: 0;
  box-shadow: 0 12px 60px rgba(111, 102, 3, 0.5);
  color: #51455c;
  font-size: 1.23em;
  font-weight: 500;
  line-height: normal;
  padding: 12px 25px;
  text-align: center;
}

.yellow-button:hover {
  background: rgb(172, 163, 64);
  box-shadow: 0 12px 60px rgba(111, 102, 3, 0.8);
  color: rgb(46, 39, 53);
}
