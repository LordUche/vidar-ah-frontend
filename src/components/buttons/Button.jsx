import React from 'react';
import styles from './button.scss';

export default function Button(props) {
  const { buttonName } = props;
  return (
    <div>
      <button className={styles.button}>
        {' '}
        {buttonName}
        {' '}
      </button>
    </div>
  );
}
