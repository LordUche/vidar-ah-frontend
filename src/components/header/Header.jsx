import React from 'react';

export default function Listing() {
  return <div>Hello World</div>;
}
