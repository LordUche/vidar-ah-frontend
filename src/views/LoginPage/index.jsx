/* eslint-disable react/jsx-no-bind */
import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { SocialIcon } from 'react-social-icons';
import {
  Container, Row, Col, Form, Alert
} from 'react-bootstrap';
import PropTypes from 'prop-types';
import Header from '../../components/header/index';
import Button from '../../components/button/Button';
import { loginUser } from '../../redux/actions/auth';
import Loader from '../../components/loader/Loader';
import './login.scss';


export const Login = ({ history, loginUser: handleLogin, loading }) => {
  const [userCredentials, setUserCredentials] = useState({
    email: null,
    password: null
  });
  const [errors, setErrors] = useState([]);

  function updateLocalState(e) {
    setUserCredentials({ ...userCredentials, [e.target.name]: e.target.value });
  }

  function renderErrors(errorsArray) {
    return errorsArray.map(error => <Alert key={error} variant="danger">{error}</Alert>);
  }

  async function onSubmit(e) {
    e.preventDefault();
    const loginRes = await handleLogin(userCredentials);
    if (!Array.isArray(loginRes)) {
      return history.push('/');
    }
    console.log(loginRes);
    return setErrors(loginRes);
  }

  return (
    <>
      <div className="purple-gradient-bg">
        {loading && <Loader />}
        <Container>
          <Header location={history.location.pathname} />
        </Container>
        <Container>
          {renderErrors(errors)}
          <Row>
            <Col md={{ span: 6, offset: 3 }} className="form-black-bg">
              <h3>Login</h3>
              <Form onSubmit={onSubmit} className="form-login">
                <Form.Control
                  type="email"
                  name="email"
                  placeholder="Email"
                  className="dark-forms"
                  required
                  onChange={updateLocalState}
                />
                <Form.Control
                  type="password"
                  name="password"
                  placeholder="Password"
                  className="dark-forms"
                  required
                  onChange={updateLocalState}
                />
                <Button text="Login" className="yellow-button-login" onClick={() => onSubmit} />
                <div className="container-extras">
                  <span>
                    <NavLink
                      to="/requestpasswordreset"
                      className="link-to-another-page"
                    >
                      Forgotten password?
                    </NavLink>
                  </span>
                </div>
              </Form>
              <div className="container-social-login">
                <span className="info">Or login using </span>
                <SocialIcon
                  url="http://twitter.com"
                  className="social-media-icons"
                />
                <SocialIcon
                  url="http://facebook.com"
                  className="social-media-icons"
                />
                <SocialIcon
                  url="http://google.com"
                  className="social-media-icons"
                />
              </div>
              <div className="container-directing-to-signup">
                <span>
                  Don't have an account? Singn up
                  <NavLink to="/signup" className="link-to-another-page">
                    here
                  </NavLink>
                </span>
              </div>
            </Col>
          </Row>
        </Container>
      </div>
    </>
  );
};

Login.propTypes = {
  history: PropTypes.object.isRequired,
  loginUser: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  loading: state.fetchReducer.fetching
});

const LoginPage = connect(
  mapStateToProps,
  { loginUser }
)(Login);

export default LoginPage;
