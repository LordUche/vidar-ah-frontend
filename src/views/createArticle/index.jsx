/* eslint-disable react/jsx-no-bind */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Container, Alert } from 'react-bootstrap';
import Create from '../../components/createForms/Create';
import Preview from '../../components/previewArticle/Preview';
import { validateArticle } from '../../utils/validator';
import { createArticle } from '../../redux/actions/articles';
import Header from '../../components/header/Header';
import './createArticle.scss';

function CreateArticle(props) {
  const [articleData, setArticleData] = useState({
    title: null,
    description: null,
    body: null
  });
  const [component, setComponent] = useState('write');
  const [loading, setLoading] = useState(false);
  const [errorsArray, setErrors] = useState([]);
  const [successMessage, setSuccessMessage] = useState('');

  async function completeArticleCreation() {
    setLoading(true);
    setErrors([]);
    const res = await props.createArticle(articleData);
    const { errors } = res;
    setLoading(false);
    if (errors) {
      setErrors(errors);
    }
    setSuccessMessage(res.message);
    props.history.push(`/articles/${res.article.slug}`);
  }

  function runValidators(e) {
    e.preventDefault();
    const errors = validateArticle(articleData);
    if (errors.length > 0) {
      setComponent('write');
      return setErrors(errors);
    }
    return completeArticleCreation();
  }

  function updateInput(e) {
    if (!(e.target.name === 'images')) {
      return setArticleData({ ...articleData, [e.target.name]: e.target.value });
    }
    return setArticleData({ ...articleData, images: e.target.files[0] });
  }

  function deleteArticle() {
    window.location = '/create-article';
  }

  function renderErrors(errors) {
    return errors.map(error => (
      <Alert key={error} variant="danger">{error}</Alert>
    ));
  }

  function renderSuccess(message) {
    return message ? <Alert variant="success">{message}</Alert> : null;
  }

  return (
    <div className="create-article-container">
      <Container>
        <Header type="purple" />
        {renderErrors(errorsArray)}
        {renderSuccess(successMessage)}
      </Container>
      <div>
        {
          component === 'write'
            ? (
              <Create
                onChangeText={updateInput}
                onSubmit={() => setComponent('preview')}
                onDelete={deleteArticle}
                article={articleData}
              />
            )
            : (
              <Preview
                onSubmit={runValidators}
                onDelete={deleteArticle}
                article={articleData}
                continueEdit={() => setComponent('write')}
                loading={loading}
              />
            )
        }
      </div>
    </div>
  );
}


CreateArticle.propTypes = {
  createArticle: PropTypes.func.isRequired
};

export default connect(() => ({}), { createArticle })(CreateArticle);
