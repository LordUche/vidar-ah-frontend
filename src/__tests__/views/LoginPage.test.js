/* eslint-env jest */
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import { Login } from '../../views/LoginPage';
import Header from '../../components/header/index';
import Button from '../../components/button/Button';


const getLoginPage = () => {
  const props = {
    history: {
      push: jest.fn(),
      location: {
        pathname: ''
      }
    },

    loginUser: jest.fn(),
    loading: false
  };


  const shallowWrapper = shallow(<Login {...props} />);

  const mountWrapper = mount(
    <MemoryRouter>
      <Login {...props} />
    </MemoryRouter>
  );
  return { shallowWrapper, mountWrapper };
};

const { mountWrapper } = getLoginPage();

describe('Test the Login Page', () => {
  it('it render the longin page without failing', () => {
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });
  it('should contain the Header', () => {
    expect(mountWrapper.find(Header)).toHaveLength(1);
    expect(mountWrapper.find(Button)).toHaveLength(1);
  });
  it('it render social media icons', () => {
    expect(mountWrapper.find('SocialIcon')).toHaveLength(3);
  });
  it('it render form inputs, buttons and links', () => {
    expect(mountWrapper.find('Container')).toHaveLength(2);
  });
});
