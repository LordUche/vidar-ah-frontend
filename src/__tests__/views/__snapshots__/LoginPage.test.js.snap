// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Login Page it render the longin page without failing 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "jirkj0",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "jirkj0",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Login
      history={
        Object {
          "location": Object {
            "pathname": "",
          },
          "push": [MockFunction],
        }
      }
      loading={false}
      loginUser={[MockFunction]}
    >
      <div
        className="purple-gradient-bg"
      >
        <Container
          as="div"
          fluid={false}
        >
          <div
            className="container"
          >
            <Header
              location=""
            >
              <Bootstrap(undefined)
                collapseOnSelect={true}
                expand="lg"
              >
                <ForwardRef
                  bsPrefix="navbar"
                  collapseOnSelect={true}
                  expand="lg"
                  innerRef={null}
                >
                  <Uncontrolled(Navbar)
                    bsPrefix="navbar"
                    collapseOnSelect={true}
                    expand="lg"
                    innerRef={null}
                  >
                    <Navbar
                      as="nav"
                      bsPrefix="navbar"
                      collapseOnSelect={true}
                      expand="lg"
                      onToggle={[Function]}
                      variant="light"
                    >
                      <nav
                        className="navbar navbar-expand-lg navbar-light"
                      >
                        <a
                          className="white-link"
                          href="#home"
                        >
                          authorsHaven
                        </a>
                        <NavbarToggle
                          aria-controls="responsive-navbar-nav"
                          as="button"
                          label="Toggle navigation"
                        >
                          <button
                            aria-controls="responsive-navbar-nav"
                            aria-label="Toggle navigation"
                            className="navbar-toggler"
                            onClick={[Function]}
                            type="button"
                          >
                            <span
                              className="navbar-toggler-icon"
                            />
                          </button>
                        </NavbarToggle>
                        <NavbarCollapse
                          id="responsive-navbar-nav"
                        >
                          <Collapse
                            appear={false}
                            dimension="height"
                            getDimensionValue={[Function]}
                            id="responsive-navbar-nav"
                            in={false}
                            mountOnEnter={false}
                            timeout={300}
                            unmountOnExit={false}
                          >
                            <Transition
                              addEndListener={[Function]}
                              appear={false}
                              aria-expanded={null}
                              enter={true}
                              exit={true}
                              id="responsive-navbar-nav"
                              in={false}
                              mountOnEnter={false}
                              onEnter={[Function]}
                              onEntered={[Function]}
                              onEntering={[Function]}
                              onExit={[Function]}
                              onExited={[Function]}
                              onExiting={[Function]}
                              timeout={300}
                              unmountOnExit={false}
                            >
                              <div
                                aria-expanded={null}
                                className="navbar-collapse collapse"
                                id="responsive-navbar-nav"
                              >
                                <Nav
                                  as="div"
                                  className="mr-auto"
                                  fill={false}
                                  justify={false}
                                >
                                  <ContextTransform(AbstractNav)
                                    as="div"
                                    className="mr-auto navbar-nav"
                                    onSelect={[Function]}
                                  >
                                    <AbstractNav
                                      as="div"
                                      className="mr-auto navbar-nav"
                                      onSelect={[Function]}
                                      parentOnSelect={[Function]}
                                    >
                                      <div
                                        className="mr-auto navbar-nav"
                                        onKeyDown={[Function]}
                                      />
                                    </AbstractNav>
                                  </ContextTransform(AbstractNav)>
                                </Nav>
                                <Nav
                                  as="div"
                                  fill={false}
                                  justify={false}
                                >
                                  <ContextTransform(AbstractNav)
                                    as="div"
                                    className="navbar-nav"
                                    onSelect={[Function]}
                                  >
                                    <AbstractNav
                                      as="div"
                                      className="navbar-nav"
                                      onSelect={[Function]}
                                      parentOnSelect={[Function]}
                                    >
                                      <div
                                        className="navbar-nav"
                                        onKeyDown={[Function]}
                                      >
                                        <span />
                                        <span />
                                        <span />
                                      </div>
                                    </AbstractNav>
                                  </ContextTransform(AbstractNav)>
                                </Nav>
                              </div>
                            </Transition>
                          </Collapse>
                        </NavbarCollapse>
                      </nav>
                    </Navbar>
                  </Uncontrolled(Navbar)>
                </ForwardRef>
              </Bootstrap(undefined)>
            </Header>
          </div>
        </Container>
        <Container
          as="div"
          fluid={false}
        >
          <div
            className="container"
          >
            <Bootstrap(Row)>
              <Row
                as="div"
                bsPrefix="row"
                noGutters={false}
              >
                <div
                  className="row"
                >
                  <Col
                    as="div"
                    className="form-black-bg"
                    md={
                      Object {
                        "offset": 3,
                        "span": 6,
                      }
                    }
                  >
                    <div
                      className="form-black-bg col-md-6 offset-md-3"
                    >
                      <h3>
                        Login
                      </h3>
                      <Form
                        as="form"
                        className="form-login"
                        inline={false}
                        onSubmit={[Function]}
                      >
                        <form
                          className="form-login"
                          onSubmit={[Function]}
                        >
                          <FormControl
                            as="input"
                            className="dark-forms"
                            name="email"
                            onChange={[Function]}
                            placeholder="Email"
                            required={true}
                            type="email"
                          >
                            <input
                              className="dark-forms form-control"
                              name="email"
                              onChange={[Function]}
                              placeholder="Email"
                              required={true}
                              type="email"
                            />
                          </FormControl>
                          <FormControl
                            as="input"
                            className="dark-forms"
                            name="password"
                            onChange={[Function]}
                            placeholder="Password"
                            required={true}
                            type="password"
                          >
                            <input
                              className="dark-forms form-control"
                              name="password"
                              onChange={[Function]}
                              placeholder="Password"
                              required={true}
                              type="password"
                            />
                          </FormControl>
                          <Button
                            text="Login"
                          >
                            <button
                              className="yellow-button yellow-button-login"
                            >
                              Login
                            </button>
                          </Button>
                          <div
                            className="container-extras"
                          >
                            <span>
                              <NavLink
                                className="link-to-another-page"
                                to="/requestpasswordreset"
                              >
                                <Route
                                  path="\\\\/requestpasswordreset"
                                >
                                  <Link
                                    aria-current={null}
                                    className="link-to-another-page"
                                    to="/requestpasswordreset"
                                  >
                                    <a
                                      aria-current={null}
                                      className="link-to-another-page"
                                      href="/requestpasswordreset"
                                      onClick={[Function]}
                                    >
                                      Forgotten password?
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                            </span>
                          </div>
                        </form>
                      </Form>
                      <div
                        className="container-social-login"
                      >
                        <span
                          className="info"
                        >
                          Or login using 
                        </span>
                        <SocialIcon
                          className="social-media-icons"
                          url="http://twitter.com"
                        >
                          <a
                            aria-label="twitter"
                            className="social-icon social-media-icons"
                            href="http://twitter.com"
                            style={
                              Object {
                                "display": "inline-block",
                                "height": "50px",
                                "overflow": "hidden",
                                "position": "relative",
                                "verticalAlign": "middle",
                                "width": "50px",
                              }
                            }
                          >
                            <div
                              className="social-container"
                              style={
                                Object {
                                  "height": "100%",
                                  "left": 0,
                                  "position": "absolute",
                                  "top": 0,
                                  "width": "100%",
                                }
                              }
                            >
                              <svg
                                className="social-svg"
                                style={
                                  Object {
                                    "borderRadius": "50%",
                                    "fillRule": "evenodd",
                                    "height": "100%",
                                    "left": 0,
                                    "position": "absolute",
                                    "top": 0,
                                    "width": "100%",
                                  }
                                }
                                viewBox="0 0 64 64"
                              >
                                <Background>
                                  <g
                                    className="social-svg-background"
                                    style={
                                      Object {
                                        "MozTransition": "fill 170ms ease-in-out",
                                        "OTransition": "fill 170ms ease-in-out",
                                        "WebkitTransition": "fill 170ms ease-in-out",
                                        "fill": "transparent",
                                        "msTransition": "fill 170ms ease-in-out",
                                        "transition": "fill 170ms ease-in-out",
                                      }
                                    }
                                  >
                                    <circle
                                      cx="32"
                                      cy="32"
                                      r="31"
                                    />
                                  </g>
                                </Background>
                                <Icon
                                  networkKey="twitter"
                                >
                                  <g
                                    className="social-svg-icon"
                                    style={
                                      Object {
                                        "MozTransition": "fill 170ms ease-in-out",
                                        "OTransition": "fill 170ms ease-in-out",
                                        "WebkitTransition": "fill 170ms ease-in-out",
                                        "fill": "transparent",
                                        "msTransition": "fill 170ms ease-in-out",
                                        "transition": "fill 170ms ease-in-out",
                                      }
                                    }
                                  >
                                    <path
                                      d="M48,22.1c-1.2,0.5-2.4,0.9-3.8,1c1.4-0.8,2.4-2.1,2.9-3.6c-1.3,0.8-2.7,1.3-4.2,1.6 C41.7,19.8,40,19,38.2,19c-3.6,0-6.6,2.9-6.6,6.6c0,0.5,0.1,1,0.2,1.5c-5.5-0.3-10.3-2.9-13.5-6.9c-0.6,1-0.9,2.1-0.9,3.3 c0,2.3,1.2,4.3,2.9,5.5c-1.1,0-2.1-0.3-3-0.8c0,0,0,0.1,0,0.1c0,3.2,2.3,5.8,5.3,6.4c-0.6,0.1-1.1,0.2-1.7,0.2c-0.4,0-0.8,0-1.2-0.1 c0.8,2.6,3.3,4.5,6.1,4.6c-2.2,1.8-5.1,2.8-8.2,2.8c-0.5,0-1.1,0-1.6-0.1c2.9,1.9,6.4,2.9,10.1,2.9c12.1,0,18.7-10,18.7-18.7 c0-0.3,0-0.6,0-0.8C46,24.5,47.1,23.4,48,22.1z"
                                    />
                                  </g>
                                </Icon>
                                <Mask
                                  networkKey="twitter"
                                >
                                  <g
                                    className="social-svg-mask"
                                    style={
                                      Object {
                                        "MozTransition": "fill 170ms ease-in-out",
                                        "OTransition": "fill 170ms ease-in-out",
                                        "WebkitTransition": "fill 170ms ease-in-out",
                                        "fill": "#00aced",
                                        "msTransition": "fill 170ms ease-in-out",
                                        "transition": "fill 170ms ease-in-out",
                                      }
                                    }
                                  >
                                    <path
                                      d="M0,0v64h64V0H0z M44.7,25.5c0,0.3,0,0.6,0,0.8C44.7,35,38.1,45,26.1,45c-3.7,0-7.2-1.1-10.1-2.9 c0.5,0.1,1,0.1,1.6,0.1c3.1,0,5.9-1,8.2-2.8c-2.9-0.1-5.3-2-6.1-4.6c0.4,0.1,0.8,0.1,1.2,0.1c0.6,0,1.2-0.1,1.7-0.2 c-3-0.6-5.3-3.3-5.3-6.4c0,0,0-0.1,0-0.1c0.9,0.5,1.9,0.8,3,0.8c-1.8-1.2-2.9-3.2-2.9-5.5c0-1.2,0.3-2.3,0.9-3.3 c3.2,4,8.1,6.6,13.5,6.9c-0.1-0.5-0.2-1-0.2-1.5c0-3.6,2.9-6.6,6.6-6.6c1.9,0,3.6,0.8,4.8,2.1c1.5-0.3,2.9-0.8,4.2-1.6 c-0.5,1.5-1.5,2.8-2.9,3.6c1.3-0.2,2.6-0.5,3.8-1C47.1,23.4,46,24.5,44.7,25.5z"
                                    />
                                  </g>
                                </Mask>
                              </svg>
                            </div>
                          </a>
                        </SocialIcon>
                        <SocialIcon
                          className="social-media-icons"
                          url="http://facebook.com"
                        >
                          <a
                            aria-label="facebook"
                            className="social-icon social-media-icons"
                            href="http://facebook.com"
                            style={
                              Object {
                                "display": "inline-block",
                                "height": "50px",
                                "overflow": "hidden",
                                "position": "relative",
                                "verticalAlign": "middle",
                                "width": "50px",
                              }
                            }
                          >
                            <div
                              className="social-container"
                              style={
                                Object {
                                  "height": "100%",
                                  "left": 0,
                                  "position": "absolute",
                                  "top": 0,
                                  "width": "100%",
                                }
                              }
                            >
                              <svg
                                className="social-svg"
                                style={
                                  Object {
                                    "borderRadius": "50%",
                                    "fillRule": "evenodd",
                                    "height": "100%",
                                    "left": 0,
                                    "position": "absolute",
                                    "top": 0,
                                    "width": "100%",
                                  }
                                }
                                viewBox="0 0 64 64"
                              >
                                <Background>
                                  <g
                                    className="social-svg-background"
                                    style={
                                      Object {
                                        "MozTransition": "fill 170ms ease-in-out",
                                        "OTransition": "fill 170ms ease-in-out",
                                        "WebkitTransition": "fill 170ms ease-in-out",
                                        "fill": "transparent",
                                        "msTransition": "fill 170ms ease-in-out",
                                        "transition": "fill 170ms ease-in-out",
                                      }
                                    }
                                  >
                                    <circle
                                      cx="32"
                                      cy="32"
                                      r="31"
                                    />
                                  </g>
                                </Background>
                                <Icon
                                  networkKey="facebook"
                                >
                                  <g
                                    className="social-svg-icon"
                                    style={
                                      Object {
                                        "MozTransition": "fill 170ms ease-in-out",
                                        "OTransition": "fill 170ms ease-in-out",
                                        "WebkitTransition": "fill 170ms ease-in-out",
                                        "fill": "transparent",
                                        "msTransition": "fill 170ms ease-in-out",
                                        "transition": "fill 170ms ease-in-out",
                                      }
                                    }
                                  >
                                    <path
                                      d="M34.1,47V33.3h4.6l0.7-5.3h-5.3v-3.4c0-1.5,0.4-2.6,2.6-2.6l2.8,0v-4.8c-0.5-0.1-2.2-0.2-4.1-0.2 c-4.1,0-6.9,2.5-6.9,7V28H24v5.3h4.6V47H34.1z"
                                    />
                                  </g>
                                </Icon>
                                <Mask
                                  networkKey="facebook"
                                >
                                  <g
                                    className="social-svg-mask"
                                    style={
                                      Object {
                                        "MozTransition": "fill 170ms ease-in-out",
                                        "OTransition": "fill 170ms ease-in-out",
                                        "WebkitTransition": "fill 170ms ease-in-out",
                                        "fill": "#3b5998",
                                        "msTransition": "fill 170ms ease-in-out",
                                        "transition": "fill 170ms ease-in-out",
                                      }
                                    }
                                  >
                                    <path
                                      d="M0,0v64h64V0H0z M39.6,22l-2.8,0c-2.2,0-2.6,1.1-2.6,2.6V28h5.3l-0.7,5.3h-4.6V47h-5.5V33.3H24V28h4.6V24 c0-4.6,2.8-7,6.9-7c2,0,3.6,0.1,4.1,0.2V22z"
                                    />
                                  </g>
                                </Mask>
                              </svg>
                            </div>
                          </a>
                        </SocialIcon>
                        <SocialIcon
                          className="social-media-icons"
                          url="http://google.com"
                        >
                          <a
                            aria-label="google"
                            className="social-icon social-media-icons"
                            href="http://google.com"
                            style={
                              Object {
                                "display": "inline-block",
                                "height": "50px",
                                "overflow": "hidden",
                                "position": "relative",
                                "verticalAlign": "middle",
                                "width": "50px",
                              }
                            }
                          >
                            <div
                              className="social-container"
                              style={
                                Object {
                                  "height": "100%",
                                  "left": 0,
                                  "position": "absolute",
                                  "top": 0,
                                  "width": "100%",
                                }
                              }
                            >
                              <svg
                                className="social-svg"
                                style={
                                  Object {
                                    "borderRadius": "50%",
                                    "fillRule": "evenodd",
                                    "height": "100%",
                                    "left": 0,
                                    "position": "absolute",
                                    "top": 0,
                                    "width": "100%",
                                  }
                                }
                                viewBox="0 0 64 64"
                              >
                                <Background>
                                  <g
                                    className="social-svg-background"
                                    style={
                                      Object {
                                        "MozTransition": "fill 170ms ease-in-out",
                                        "OTransition": "fill 170ms ease-in-out",
                                        "WebkitTransition": "fill 170ms ease-in-out",
                                        "fill": "transparent",
                                        "msTransition": "fill 170ms ease-in-out",
                                        "transition": "fill 170ms ease-in-out",
                                      }
                                    }
                                  >
                                    <circle
                                      cx="32"
                                      cy="32"
                                      r="31"
                                    />
                                  </g>
                                </Background>
                                <Icon
                                  networkKey="google"
                                >
                                  <g
                                    className="social-svg-icon"
                                    style={
                                      Object {
                                        "MozTransition": "fill 170ms ease-in-out",
                                        "OTransition": "fill 170ms ease-in-out",
                                        "WebkitTransition": "fill 170ms ease-in-out",
                                        "fill": "transparent",
                                        "msTransition": "fill 170ms ease-in-out",
                                        "transition": "fill 170ms ease-in-out",
                                      }
                                    }
                                  >
                                    <path
                                      d="M35.4,17h-8c-1.1,0-2.2,0.1-3.4,0.4 c-1.2,0.3-2.4,0.9-3.5,1.8c-1.7,1.6-2.5,3.4-2.5,5.4c0,1.6,0.6,3.1,1.8,4.3c1.1,1.3,2.7,2,4.9,2c0.4,0,0.8,0,1.3-0.1 c-0.1,0.2-0.2,0.4-0.2,0.7c-0.1,0.2-0.2,0.5-0.2,0.9c0,0.6,0.1,1.1,0.4,1.5c0.2,0.4,0.5,0.8,0.8,1.2c-0.9,0-2.1,0.1-3.5,0.4 c-1.4,0.2-2.8,0.7-4.1,1.5c-1.2,0.7-1.9,1.5-2.4,2.4c-0.5,0.9-0.7,1.7-0.7,2.5c0,1.5,0.7,2.8,2.1,3.9c1.4,1.2,3.5,1.8,6.3,1.8 c3.3-0.1,5.9-0.9,7.7-2.4c1.7-1.5,2.6-3.2,2.6-5.2c0-1.4-0.3-2.5-0.9-3.3c-0.6-0.8-1.4-1.6-2.2-2.3l-1.4-1.1 c-0.2-0.2-0.4-0.4-0.6-0.7c-0.2-0.3-0.4-0.6-0.4-1c0-0.4,0.1-0.8,0.4-1.1c0.2-0.3,0.4-0.6,0.7-0.8c0.4-0.4,0.8-0.7,1.2-1.1 c0.3-0.4,0.6-0.7,0.9-1.2c0.6-0.9,0.9-2,0.9-3.4c0-0.8-0.1-1.5-0.3-2.1c-0.2-0.6-0.5-1.1-0.7-1.5c-0.3-0.5-0.6-0.8-0.9-1.2 c-0.3-0.3-0.6-0.5-0.8-0.7H33L35.4,17z M31,38.9c0.7,0.8,1,1.6,1,2.7c0,1.3-0.5,2.3-1.5,3.1c-1,0.8-2.4,1.2-4.3,1.3 c-2.1,0-3.8-0.5-5-1.4c-1.3-0.9-1.9-2.1-1.9-3.5c0-0.7,0.1-1.3,0.4-1.8c0.3-0.5,0.6-0.9,0.9-1.2c0.4-0.3,0.8-0.6,1.1-0.7 c0.4-0.2,0.7-0.3,0.9-0.4c0.9-0.3,1.7-0.5,2.5-0.6c0.8-0.1,1.4-0.1,1.6-0.1c0.3,0,0.6,0,0.9,0C29.2,37.3,30.3,38.2,31,38.9z  M29.7,27.1c-0.1,0.5-0.3,1.1-0.7,1.6c-0.7,0.7-1.6,1.1-2.6,1.1c-0.8,0-1.6-0.3-2.2-0.8c-0.6-0.5-1.2-1.1-1.6-1.9 c-0.8-1.6-1.3-3.1-1.3-4.5c0-1.1,0.3-2.1,0.9-3c0.7-0.9,1.6-1.3,2.7-1.3c0.8,0,1.5,0.3,2.2,0.7c0.6,0.5,1.1,1.1,1.5,1.9 c0.8,1.6,1.2,3.2,1.2,4.8C29.8,26.1,29.8,26.5,29.7,27.1z M43.7,29.5v-4.3h-2.5v4.3H37V32h4.2v4.2h2.5V32H48v-2.5H43.7z"
                                    />
                                  </g>
                                </Icon>
                                <Mask
                                  networkKey="google"
                                >
                                  <g
                                    className="social-svg-mask"
                                    style={
                                      Object {
                                        "MozTransition": "fill 170ms ease-in-out",
                                        "OTransition": "fill 170ms ease-in-out",
                                        "WebkitTransition": "fill 170ms ease-in-out",
                                        "fill": "#dd4b39",
                                        "msTransition": "fill 170ms ease-in-out",
                                        "transition": "fill 170ms ease-in-out",
                                      }
                                    }
                                  >
                                    <path
                                      d="M0,0v64h64V0H0z M31.3,19.1c0.3,0.3,0.6,0.7,0.9,1.2c0.3,0.4,0.5,0.9,0.7,1.5c0.2,0.6,0.3,1.3,0.3,2.1 c0,1.4-0.3,2.6-0.9,3.4c-0.3,0.4-0.6,0.8-0.9,1.2c-0.4,0.4-0.8,0.7-1.2,1.1c-0.2,0.2-0.5,0.5-0.7,0.8c-0.2,0.3-0.4,0.7-0.4,1.1 c0,0.4,0.1,0.8,0.4,1c0.2,0.3,0.4,0.5,0.6,0.7l1.4,1.1c0.8,0.7,1.6,1.5,2.2,2.3c0.6,0.8,0.9,2,0.9,3.3c0,1.9-0.9,3.7-2.6,5.2 c-1.8,1.6-4.3,2.4-7.7,2.4c-2.8,0-4.9-0.6-6.3-1.8c-1.4-1.1-2.1-2.4-2.1-3.9c0-0.7,0.2-1.6,0.7-2.5c0.4-0.9,1.2-1.7,2.4-2.4 c1.3-0.7,2.7-1.2,4.1-1.5c1.4-0.2,2.6-0.3,3.5-0.4c-0.3-0.4-0.5-0.8-0.8-1.2c-0.3-0.4-0.4-0.9-0.4-1.5c0-0.4,0-0.6,0.2-0.9 c0.1-0.2,0.2-0.5,0.2-0.7c-0.5,0.1-0.9,0.1-1.3,0.1c-2.1,0-3.8-0.7-4.9-2c-1.2-1.2-1.8-2.7-1.8-4.3c0-2,0.8-3.8,2.5-5.4 c1.1-0.9,2.3-1.6,3.5-1.8c1.2-0.2,2.3-0.4,3.4-0.4h8L33,18.4h-2.5C30.7,18.6,31,18.8,31.3,19.1z M48,32h-4.3v4.2h-2.5V32H37v-2.5 h4.2v-4.3h2.5v4.3H48V32z M27.1,19.1c-0.6-0.5-1.4-0.7-2.2-0.7c-1.1,0-2,0.5-2.7,1.3c-0.6,0.9-0.9,1.9-0.9,3c0,1.5,0.4,3,1.3,4.5 c0.4,0.7,0.9,1.4,1.6,1.9c0.6,0.5,1.4,0.8,2.2,0.8c1.1,0,1.9-0.4,2.6-1.1c0.3-0.5,0.6-1,0.7-1.6c0.1-0.5,0.1-1,0.1-1.4 c0-1.6-0.4-3.2-1.2-4.8C28.2,20.2,27.7,19.5,27.1,19.1z M26.9,36.2c-0.2,0-0.7,0-1.6,0.1c-0.8,0.1-1.7,0.3-2.5,0.6 c-0.2,0.1-0.5,0.2-0.9,0.4c-0.4,0.2-0.7,0.4-1.1,0.7c-0.4,0.3-0.7,0.7-0.9,1.2c-0.3,0.5-0.4,1.1-0.4,1.8c0,1.4,0.6,2.6,1.9,3.5 c1.2,0.9,2.9,1.4,5,1.4c1.9,0,3.3-0.4,4.3-1.3c1-0.8,1.5-1.8,1.5-3.1c0-1-0.3-1.9-1-2.7c-0.7-0.7-1.8-1.6-3.3-2.6 C27.5,36.2,27.2,36.2,26.9,36.2z"
                                    />
                                  </g>
                                </Mask>
                              </svg>
                            </div>
                          </a>
                        </SocialIcon>
                      </div>
                      <div
                        className="container-directing-to-signup"
                      >
                        <span>
                          Don't have an account? Singn up
                          <NavLink
                            className="link-to-another-page"
                            to="/signup"
                          >
                            <Route
                              path="\\\\/signup"
                            >
                              <Link
                                aria-current={null}
                                className="link-to-another-page"
                                to="/signup"
                              >
                                <a
                                  aria-current={null}
                                  className="link-to-another-page"
                                  href="/signup"
                                  onClick={[Function]}
                                >
                                  here
                                </a>
                              </Link>
                            </Route>
                          </NavLink>
                        </span>
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </Bootstrap(Row)>
          </div>
        </Container>
      </div>
    </Login>
  </Router>
</MemoryRouter>
`;
